### PR TITLE
Add Initial Attachments_Utils

### LINKED ISSUE #311: Add a the ability to add buttons to a message
Good example of call api required - https://github.com/slackapi/python-link-button-example/blob/master/example.py. Extended to work a bit nicer.

Additionally, will likely be similar to post_message with additional specification for the button(s)

### PR DESCRIPTION
**Pull request by Guthers**
Contains the framework (with some examples at the bottom) for adding json valid object to the post message call

Fixed #311

### CODE REVIEW

**Comment by mitchmcdee** [comment]
before I review this, can you pls run a linter on it to ensure its PEP-8 compliant

File: uqcsbot/util/attachments_util.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+    def validate(self) -> bool:
+        if not self.attachment_title_link == None:
+            if self.attachment_title == None:
+                # Should only define the link when the title is set
+                return False
+            if not validateURL(self.attachment_title_link):
+                # Attachment Link should be valid if not None
+                return None
+        if not self.attachment_fields == []:
```
Comment (TRManderson) [inline uqcsbot/util/attachments_util.py]: `!=` instead of `not ... ==`

File: uqcsbot/util/attachments_util.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    def __init__(self, fallback:str, text:str):
+        self.attachment_fallback = fallback
+        self.attachment_text = text
+
+    def set_color(self, ac:AttachmentColor):
+        self.attachment_color = ac.value
+
+    def validate(self) -> bool:
+        if not self.attachment_title_link == None:
+            if self.attachment_title == None:
```
Comment (TRManderson) [inline uqcsbot/util/attachments_util.py]: comparisons with `None` should use `is` and `is not`

File: uqcsbot/util/attachments_util.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    COLOUR_WARNING = "warning"
+    COLOUR_DANGER = "danger"
+
+# https://api.slack.com/docs/message-attachments
+class Attachment:
+    # Non Optional Params (can be "")
+    attachment_fallback:str
+    attachment_text:str
+
+    # Optional Params
```
Comment (TRManderson) [inline uqcsbot/util/attachments_util.py]: Please mark optional params using the `Optional` type

File: uqcsbot/util/attachments_util.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+class ActionType(Enum):
+    TYPE_BUTTON = "button"
+
+class AttachmentAction:
+    action_type:str
+    action_text:str
+    
+
+    # Optional Depending on if it is used in the action type
```
Comment (TRManderson) [inline uqcsbot/util/attachments_util.py]: Mark optionals with `Optional`

File: uqcsbot/util/attachments_util.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    action_url:str = None
+    # 
+    action_name:str = None
+    action_value:str = None
+    
+    
+    button_style:str = None
+
+    def validate(self):
+        if not self.action_url == None:
```
Comment (TRManderson) [inline uqcsbot/util/attachments_util.py]: comparison with none - use `is`

File: uqcsbot/util/attachments_util.py [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+            # Validates each of the actions to make sure they are valid
+            act:AttachmentAction
+            for act in self.attachment_actions:
+                if not act.validate:
+                    return False
+
+        return True
+    
+
+class Attachments_Util:
```
Comment (TRManderson) [inline uqcsbot/util/attachments_util.py]: Convention in Python is no underscores, upper camel case for class names. Also please explicitly specify subclassing from `object`.

**Comment by mitchmcdee** [comment]
as per irl discussion, will review this after addressing tom's comments 😗 
make sure to run `tox`!

**Comment by nicklambourne** [comment]
@Guthers Will close this for now, I think the attachments API has changed since, as well. Let me know if you want to have another go.

**Pull request closed without merging**